### PR TITLE
Added models for settings

### LINKED ISSUE #937: Add models for Settings
**I'm submitting a ...**
- [ ] bug report
- [x] feature request

**Current behaviour:**
There is no such model is present

**Expected behaviour:**
There should be model for storing links and cloud services keys

**i am working on it**

### PR DESCRIPTION
**Pull request by vaibhavsingh97**
Fixes #937 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:
Added model to store app name, app description, secret key, Storage options and Secret keys associated with it, Social login credentials, payment credentials, email settings, analytics id keys and social links

### CODE REVIEW

**Comment by vaibhavsingh97** [comment]
Cc @djmgit @ParthS007 @gabru-md @yashLadha

File: api/models/settings.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
+from api.db import db
+
+
+class ApplicationEnvironament:
```
Comment (yashLadha) [inline api/models/settings.py]: Why to have this class when you can populate some default data and give the user the option from the options, like a drop down.

File: api/models/settings.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+    PRODUCTION = 'production'
+    TESTING = 'testing'
+
+
+class Settings(db.Model):
+    __tablename__ = 'settings'
+
+    app_environment = db.Column(db.String, default=applicationEnvironament.PRODUCTION)
+    # Name of the application. (Eg. Badgeyay)
+    app_name = db.Column(db.String)
```
Comment (yashLadha) [inline api/models/settings.py]: Will it be of any advantage??

File: api/models/settings.py [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+    # Description of the App
+    app_description = db.Column(db.String(140))
```
Comment (yashLadha) [inline api/models/settings.py]: Where do you think it will be beneficial to store, as if you also use the description in accessing keys from gateway them also you have to either write or manually copy paste the thing

File: api/models/settings.py (hunk 1 above)
Comment (vaibhavsingh97) [reply api/models/settings.py]: It will give user a option, PLease look at L17, I already added that, so it's more like a radio button

File: api/models/settings.py (hunk 2 above)
Comment (vaibhavsingh97) [reply api/models/settings.py]: Its advantage is by chance, if admin wants to change something he doesn't have to look over the code and rename things, All changes at one place

File: api/models/settings.py (hunk 3 above)
Comment (vaibhavsingh97) [reply api/models/settings.py]: @yashLadhaI am not clear what you were asking? I added this which we directly populate the description box above `create badges`

File: api/models/settings.py [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+    DEVELOPMENT = 'development'
+    STAGING = 'staging'
+    PRODUCTION = 'production'
+    TESTING = 'testing'
+
+
+class Settings(db.Model):
+    __tablename__ = 'settings'
+
+    app_environment = db.Column(db.String, default=ApplicationEnvironament.PRODUCTION)
```
Comment (yashLadha) [inline api/models/settings.py]: It won't be a radio button as the user can send any malicious string to the backend. Validators should validate the input from the given set of config.

File: api/models/settings.py [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+    # -------------
+    # Social Login
+    # -------------
+
+    # Google Auth is handled by Firebase
+    # FB auth
+    fb_client_id = db.Column(db.String)
+    fb_client_secret = db.Column(db.String)
+    # Twitter auth
+    tw_consumer_key = db.Column(db.String)
```
Comment (gabru-md) [inline api/models/settings.py]: Why are these needed?

File: api/models/settings.py [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+    tw_consumer_secret = db.Column(db.String)
+
+    # -------------
+    # Payment Gateway
+    # -------------
+
+    # Stripe Keys
+    stripe_client_id = db.Column(db.String)
```
Comment (gabru-md) [inline api/models/settings.py]: What is this? Please give a strong reason to justify the changes.

File: api/models/settings.py [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+    # -------------
+
+    # Google Analytics
+    analytics_key = db.Column(db.String)
+
+    # -------------
+    # Social links
+    # -------------
+
+    support_url = db.Column(db.String)
```
Comment (gabru-md) [inline api/models/settings.py]: What is this? Please provide with a reason

File: api/models/settings.py [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+    PRODUCTION = 'production'
+    TESTING = 'testing'
+
+
+class Settings(db.Model):
+    __tablename__ = 'settings'
+
+    app_environment = db.Column(db.String, default=ApplicationEnvironament.PRODUCTION)
+    # Name of the application. (Eg. Badgeyay)
+    app_name = db.Column(db.String)
```
Comment (gabru-md) [inline api/models/settings.py]: Why are you having a variable named `app_name` ?

File: api/models/settings.py [hunk 9]
```diff
@@ -0,0 +1,237 @@
...
+    secret = db.Column(db.String)
+
+    # -------------
+    # Storage
+    # -------------
+
+    # storage place local, s3,etc
+    storage_place = db.Column(db.String)
+    # S3
+    aws_key = db.Column(db.String)
```
Comment (gabru-md) [inline api/models/settings.py]: s3 is not being used. why keep it here?

**Comment by gabru-md** [comment]
please close this @djmgit @ParthS007
